server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -0,0 +1,96 @@
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Standard format that has only content for writes. Read interface does not exist as it not yet required. This format
+ * should be used for writing data from in-memory to remote store where there is no need for checksum and the client
+ * library for the remote store has inbuilt checksum capabilities while upload and download both. This format would
+ * serialise the data in Json format and store it on remote store as is. This does not support compression yet (this
+ * can be changed as required). In comparison to {@link ChecksumBlobStoreFormat}, this format does not add any additional
+ * metadata (like header and footer) to the content. Hence, this format does not depend on {@code CodecUtil} from
+ * Lucene library.
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreFormat<T extends ToXContent> {
+
+    private final String blobNameFormat;
+
+    /**
+     * @param blobNameFormat format of the blobname in {@link String#format} format
+     */
+    public BlobStoreFormat(String blobNameFormat) {
+        this.blobNameFormat = blobNameFormat;
+    }
+
+    private String blobName(String name) {
+        return String.format(Locale.ROOT, blobNameFormat, name);
+    }
+
+    private BytesReference serialize(final T obj) throws IOException {
+        try (XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()) {
+            xContentBuilder.startObject();
+            obj.toXContent(xContentBuilder, ToXContent.EMPTY_PARAMS);
+            xContentBuilder.endObject();
+            return BytesReference.bytes(xContentBuilder);
+        }
+    }
+
+    public void writeAsyncWithUrgentPriority(T obj, BlobContainer blobContainer, String name, ActionListener<Void> listener)
+        throws IOException {
+        if (blobContainer instanceof AsyncMultiStreamBlobContainer == false) {
+            write(obj, blobContainer, name);
+            listener.onResponse(null);
+            return;
+        }
+        String blobName = blobName(name);
+        BytesReference bytes = serialize(obj);
+        String resourceDescription = "BlobStoreFormat.writeAsyncWithPriority(blob=\"" + blobName + "\")";
+        try (IndexInput input = new ByteArrayIndexInput(resourceDescription, BytesReference.toBytes(bytes))) {
+            try (
+                RemoteTransferContainer remoteTransferContainer = new RemoteTransferContainer(
+                    blobName,
+                    blobName,
+                    bytes.length(),
+                    true,
+                    WritePriority.URGENT,
+                    (size, position) -> new OffsetRangeIndexInputStream(input, size, position),
+                    null,
+                    ((AsyncMultiStreamBlobContainer) blobContainer).remoteIntegrityCheckSupported()
```

Comment:
As we are passing null checksum, `remoteIntegrityCheckSupported` should be passed as false, right?

Reply:
Actually, going by the field name, it does support the remote integrity, however, we don't need to supply the checksum since it would be validated by default by the client library.